### PR TITLE
I18n sync loads the AI lab UI and datasets-manifest.json strings

### PR DESCRIPTION
**Pull request by daynew**
AI Lab has translatable content in `datasets-manifest.json` and `ailab.json`, so these files need to be added to the I18n sync.
* Extracts the display `name` for each dataset from `datasets-manifest.json` and adds it to the JSON uploaded to Crowdin.
* Adds `ailab.json` to the I18n sync which contains the UI strings in AI Lab.

## Links
* JIRA: [FND-2121](https://codedotorg.atlassian.net/browse/FND-2121)/[P20-13](https://codedotorg.atlassian.net/browse/P20-13)
* The new `datasets-manifest.json` PR: https://github.com/code-dot-org/ml-playground/pull/298

## Testing story
* Manually ran a local sync-in to make sure the files were processed as expected.
Screenshot 2022-12-03 at 00 29 54

## Deployment strategy
* https://github.com/code-dot-org/ml-playground/pull/298 and be published.
* The code-dot-org/code-dot-org repo needs to have the `ml-playground` npm dependency updated after the above PR is published.

### CODE REVIEW

File: bin/i18n-codeorg/in.sh [hunk 1]
```diff
@@ -56,8 +56,12 @@ done
 ### Oceans tutorial
 cp_in apps/node_modules/@code-dot-org/ml-activities/i18n/oceans.json i18n/locales/source/blockly-mooc/fish.json
 
-### AI Lab datasets
+### AI Lab UI Strings
+source_ui=apps/node_modules/@code-dot-org/ml-playground/i18n/ailab.json
+staged_ui=i18n/locales/source/blockly-mooc/ailab.json
```
Comment (mgc1194) [inline bin/i18n-codeorg/in.sh]: The [apps/i18n/ailab/en_us.json](https://github.com/code-dot-org/code-dot-org/blob/425b122757673e82564bc1968300b0c01490b902/apps/i18n/ailab/en_us.json) gets sync in with the following command as well:
`### Apps
orig_dir=apps/i18n
loc_dir=i18n/locales/source/blockly-mooc
mkdir -p $loc_dir`
Will this override changes if strings are added to the the json file in the apps directory?

**Review by wilkie — APPROVED** [review]
I am familiar with this code after pairing on it and it looks good! It unifies the ml-playground localizations into our normal sync and allows these to be separate localizations that are imported and passed along with an AiLab level is rendered.

Greak work. 👍

File: bin/i18n-codeorg/in.sh [hunk 2]
```diff
@@ -56,8 +56,12 @@ done
 ### Oceans tutorial
 cp_in apps/node_modules/@code-dot-org/ml-activities/i18n/oceans.json i18n/locales/source/blockly-mooc/fish.json
 
-### AI Lab datasets
+### AI Lab UI Strings
+source_ui=apps/node_modules/@code-dot-org/ml-playground/i18n/mlPlayground.json
+staged_ui=i18n/locales/source/external-sources/ml-playground/mlPlayground.json
```
Comment (mgc1194) [inline bin/i18n-codeorg/in.sh:61]: `ml-playgorund` strings will be located in `i18n/locales/source/external-sources/ml-playground` in order to keep modularity.

File: bin/i18n/sync-out.rb [hunk 3]
```diff
@@ -392,21 +392,31 @@ def distribute_translations(upload_manifests)
...
+    Dir.glob("#{locale_dir}/external-sources/ml-playground/mlPlayground.json") do |loc_file|
+      relative_path = loc_file.delete_prefix(locale_dir)
+      next unless file_changed?(locale, relative_path)
+
+      basename = File.basename(loc_file, '.json')
+      destination = "apps/i18n/#{basename}/#{js_locale}.json"
+      sanitize_file_and_write(loc_file, destination)
+    end
+
     ### Merge ml-playground datasets into apps' ailab JSON
```
Comment (mgc1194) [inline bin/i18n/sync-out.rb]: I need to change this. We dont merge with ailab JSON anymore.

File: dashboard/app/helpers/levels_helper.rb [hunk 4]
```diff
@@ -413,6 +413,7 @@ def render_app_dependencies
     use_weblab = @level.game == Game.weblab
     use_phaser = @level.game == Game.craft
     use_javalab = @level.is_a?(Javalab)
+    use_ailab = @level.is_a?(Ailab)
```
Comment (mgc1194) [inline dashboard/app/helpers/levels_helper.rb:416]: A new helper is required to identify Ai lab levels.

File: dashboard/app/views/levels/_apps_dependencies.html.haml [hunk 5]
```diff
@@ -55,6 +56,10 @@
     %script{src: webpack_asset_path('js/blockly.js')}
     %script{src: webpack_asset_path("js/#{js_locale}/blockly_locale.js")}
 
+-# ailab external repo dependencies
+- if use_ailab
+  %script{src: webpack_asset_path("js/#{js_locale}/mlPlayground_locale.js")}
```
Comment (mgc1194) [inline dashboard/app/views/levels/_apps_dependencies.html.haml:61]: In `ailab` levels, it is necessary to add `mlPlayground_locale.js` to make translations available. By default, only the `#{app}_locale.js` is added.
Therfore for ailab levels, `ailab_locale.js` and `mlPlayground_locale.js` will be available.

File: apps/i18n/mlPlayground/en_us.json [hunk 6]
```diff
@@ -0,0 +1,3430 @@
+{
```
Comment (mgc1194) [inline apps/i18n/mlPlayground/en_us.json:1]: No translations should be added to this file. New translations must be added `ml-playground` repo.
This file is needed for apps to work, but it does not need to have any content. AI Oceans repo work the same way.

**Review by daynew — COMMENTED** [review]
https://github.com/code-dot-org/code-dot-org/pull/49218/commits/9eef400b370a1a49a7c01a65aa898c9b21714491
This Commit should be reverted. "type" is not a translated field.

**Pull request merged**